% !Mode:: "TeX:UTF-8"
% !TEX program  = xelatex
\section{Conclusion}
Dimensionality reduction is a crucial step for RNA-seq data analysis. It is a necessary upstream step for cell clustering and trajectory inference. In this study, we proposed a new method for reducing the dimensionality of single cell sequencing data: ZIVA. We tested the performance of ZIVA on several datasets and compared it with six other methods. The result shows that ZIVA can achieve better results in some cases. Some popular methods focus on the local structure but cannot preserve global structure well such as tSNE. However, tSNE may fails when data are continuous distributed. Some methods try to keep global structure by maximize the explainable variations but fail to preserve subtle local structures such as PCA. ZIVA have a good balance between local structure and global structure. The result shows that ZIVA can recover the subtle batch effects and can also well recover the cell developmental lineage.\\
A shortcoming of ZIVA is the relatively long running time. For the large datasets that contains thousands of cells, ZIVA may cost several hours on a desktop-level computer. It would be better if we can compress the model to reduce the time costing.
