% !Mode:: "TeX:UTF-8"
% !TEX program  = xelatex
\section{Introduction}
The next-generation sequencing technologies are rapidly growing. Single-cell RNA sequencing (scRNA-seq) enable researchers to get the gene expression level of thousands of cells at the same time. It can greatly aid to identify molecular heterogeneity and biological mechanisms at the single cellular level \cite{shapiro2013single}.\\
However, the single cell sequencing data usually have more than two thousand dimensions which makes it difficult to be analyzed. Many genes are expressed at the same time. They are well regulated and interacted. Therefore, there are only limited number of cell states in a group of cells. Finding a meaningful low dimensional representation to represent those cell states is a necessary step in scRNA-seq data analyzing. It is assumed that those data form a low dimensional manifold in the high dimensional space. The process of dimensionality reduction extracts crucial information in the original high dimensional data and represent them in a low dimensional structure.  It plays an important role in data visualization and data preprocessing which allows efficient downstream analysis such as trajectory inference \cite{Saelens2019}, cell clustering \cite{weber2016comparison} and cell sub-population identification \cite{Hwang2018}. For example, Seurat uses PCA \cite{Abdi2010} to reduce dimensionality before clustering \cite{Satija2015}. Destiny \cite{angerer2016destiny} uses diffusion map before clustering. Monocle \cite{Qiu2017} performs uniform manifold approximation and projection (UMAP) \cite{McInnes2018} or independent component analysis (ICA) \cite{hyvarinen2000independent} before inferencing cell trajectories. \\
There are many traditional dimensionality reduction methods that work well on the majority of datasets. Some of them seek to preserve global structure such as PCA \cite{Abdi2010} and MDS \cite{Kruskal1964}, while others prefer to keep local structures such as LLE \cite{roweis2000nonlinear} and tSNE \cite{Kobak2019}. However, itâ€™s always a challenge to preserve global structure and the local structure at the same time. \\ 
Those dimensionality methods work well for bulk cell data. However, the single cell sequencing data have more noise than bulk cell data because of the low capture efficiency in single cell sequencing \cite{Peng2019}. Thus, single cell data have excessive dropout events which means many data points that should have bigger value are presented as zero or near zero. The traditional methods are inefficient because of those noises.  \\
To address this challenge, many methods that are specific to single cell expression data are developed. There are three main categories. The first strategy is to probabilistically model the conditional dropout events such as ZIFA \cite{Pierson2015}. It uses a double exponential function to mimic the dropout probability and uses EM algorithm \cite{mclachlan2007algorithm} to identify the model parameters.
The second strategy is to impute those dropouts by borrowing values of the same gene in other cells. Those cells are chose based on the genes that are unlikely to be affected by dropout events. For example, RESCUE \cite{Tracy2019} and scImpute \cite{Li2018}. \\
Moreover, one kind of approaches make use of deep learning technologies. They build the model based on autoencoder or variational autoencoder and use the bottleneck layer as the dimensionality reduction results. For example, VASC \cite{Wang2018} adds the dropout model in ZIFA to a VAE model to enable it to model the dropout events. DCA \cite{Eraslan2019a} uses autoencoder to impute the missing values by assuming the zero-inflated negative binomial model \cite{Hafemeister2019}. scScope \cite{Deng2019} uses a recurrent neuron network to impute and reduce the dimensionality. \\
In this study, I compared the performance of several popular dimensionality reduction methods on scRNA-seq data. Also, based on the structure of VAE \cite{Kingma2014} and VASC \cite{Wang2018}, I developed a model, zero-inflated variational autoencoder (ZIVA) to analyze and visualize the single cell RNA sequencing data. It uses the gumbel softmax and double exponential function to model the dropout events. I compared the performance of ZIVA and some other popular methods and showed that ZIVA outperforms some methods and have better datasets compatibility.

